content/core-features/authentication.textile

```diff
@@ -373,6 +374,125 @@ The following JavaScript example demonstrates how to issue an "Ably Token":#ably
+- allowReauthMargin := The @allowReauthMargin@ bool permits a token renewal cycle to take place without needing established connections to be dropped, by postponing enforcement to 30 seconds in the future, and sending any existing connections a hint to obtain (and upgrade the connection to use) a new token. It defaults to @false@, meaning that the effect is near-immediate.
+
+When invoking the revocation API, a client must prove possession of the API key in question. This can be done by using basic authentication, using the API key itself.
+
+There can be a maximum of 100 target specifiers in a single request.
+
+The token revocation API is rate-limited, so there is a maximum global aggregate rate of revocation requests per app. The rate is configurable by Ably at the application level. As part of the process of enabling revocation on an app Ably will ask you what the maximum rate of token revocations on that app needs to be, and then provision that capacity. The decision is not permanent, it can be changed by Ably on request at any time.
+
+Note that revocation is not a permanent prohibition of any client, connection, @clientId@ or channel. It invalidates token credentials issued before a certain point in time. Ultimately, the app that issues tokens is the arbiter of whether or not a client is able to continue to interact with the Ably service.
+
+It may be that if you enable token revocation, and attempt to revoke a token already created earlier, then that token cannot be guaranteed to be revoked if it has already been cached.
+
+h4. Curl examples
+
+A sample request to revoke an Ably token based on @clientId@:
+
+```[sh]
+curl -u "${ABLY_KEY}:${ABLY_SECRET}" -X POST https://rest.ably.io/keys/${ABLY_KEY}/revokeTokens -d @data1.json \
+     --header "Accept: application/json" \
+     --header "Content-Type: application/json"
+```
+
+Where @data.json@ is as follows:
+
+```[json]
+{
+    "targets": [ "clientId:client1@example.com" ]
+}
+```
+
+A sample request to revoke a JWT based on @revocationKey@:
+
+```[sh]
+curl -u "${ABLY_KEY}:${ABLY_SECRET}" -X POST https://rest.ably.io/keys/${ABLY_KEY}/revokeTokens -d @data1.json \
+     --header "Accept: application/json" \
+     --header "Content-Type: application/json"
+```
+
+Where @data.json@ is as follows:
+
+```[json]
+{
+    "targets": [ "revocationKey:client1@example.com" ]
+}
+```
+
+h3. Revocation errors
+
+System error responses arising from the use of an invalidated token will include an Ably @ErrorInfo@ with a @code@ of 40146 and an informative error message. This includes REST API error responses, and Ably connection @disconnected@ state changes. Such a token error is non-fatal, which means that client libraries can react to that error by triggering a token renewal where the given library options allow (that is, if an @authCallback@ or @authURL@ is specified).
```

Comment:
40141, not 40146 - https://github.com/ably/ably-common/blob/main/protocol/errors.json#L55

Reply:
Changed to 40141